File: api_nestjs_distributed_order_platform/src/presentation/http/mappers/payment.mapper.ts
import { PaymentEntity } from "src/domain/entities/payment.entity";

export function toPaymentResponse(payment: PaymentEntity) {
    return {
        id: payment.getId(),
        orderId: payment.getOrderId(),
        amount: payment.getAmount(),
        status: payment.getStatus(),
        attempts: payment.getAttempts(),
        externalId: payment.getExternalId(),
        lastErrorCode: payment.getLastErrorCode(),
        lastErrorMessage: payment.getLastErrorMessage(),
    };
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/presentation/http/mapper/PaymentMapper.java
package com.project.order.presentation.http.mapper;

import com.project.order.domain.model.Payment;
import com.project.order.presentation.http.dto.PaymentDto.PaymentResponse;

public class PaymentMapper {

    private PaymentMapper() {}

    public static PaymentResponse toResponse(Payment payment) {
        return new PaymentResponse(
                payment.getId(),
                payment.getOrderId(),
                payment.getAmount().getValue(),
                payment.getStatus().name(),
                payment.getAttempts(),
                payment.getLastErrorCode(),
                payment.getLastErrorMessage(),
                payment.getExternalId()
        );
    }
}
